Source/Parser.swift

```diff
@@ -182,17 +92,17 @@ public class Parser {
 
         - returns: The result Dictionary.
     */
-    public class func parseProperties(json: Any, closure: (ParserPropertyMaker) -> Void) throws -> [String: Any] {
+    public class func parseEntity(json: Any, closure: (ParserPropertyMaker) -> Void) throws -> [String: Any] {
         if let json = json as? [String: Any] {
-            return try parsePropertiesForJSONDictionary(json, closure: closure)
+            return try parseEntityForJSONDictionary(json, closure: closure)
         } else if let json = json as? [Any] {
-            return try parsePropertiesForJSONArray(json, closure: closure)
+            return try parseEntityForJSONArray(json, closure: closure)
         } else {
             throw ParserError.validation(failureReason: "JSON object was not of type: [String: Any] or [Any]")
         }
     }
 
-    private class func parsePropertiesForJSONDictionary(_ dictionary: [String: Any], closure: (ParserPropertyMaker) -> Void) throws -> [String: Any] {
+    private class func parseEntityForJSONDictionary(_ dictionary: [String: Any], closure: (ParserPropertyMaker) -> Void) throws -> [String: Any] {
```

Comment:
Also, `ParserPropertyMaker` probably needs to be refactored at this point if we're going to referring to it as a `schema` in the closure.